API.adoc

```diff
@@ -0,0 +1,402 @@
+=== SMART COSMOS Things Edge API Documentation
+
+The SMART COSMOS Things Edge API provides a standard set of CRUD methods for
+creating,reading, updating, and deleting a Thing and its associated Metadata
+in a single operation. Each CRUD action requires that the User executing the
+call has at least one Role with the appropriate Authority or Authorities. The
+Authorities relevant to the Things Edge API are listed below.
+
+== Authorities
+
+* https://authorities.smartcosmos.net/metadata/create
+* https://authorities.smartcosmos.net/metadata/update
+* https://authorities.smartcosmos.net/metadata/read
+* https://authorities.smartcosmos.net/metadata/delete
+* https://authorities.smartcosmos.net/things/create
+* https://authorities.smartcosmos.net/things/update
+* https://authorities.smartcosmos.net/things/read
+* https://authorities.smartcosmos.net/things/delete
+
+*Available methods*
+
+* <<create1, Create - `POST /{type}`>>
+* <<update1, Update - `PUT /{type}/{urn}`>>
+* <<read1, Read - `GET /{type}?fields={keyNames}&page={pageNumber}&size={pageSize}`>>
+* <<read2, Read - `GET /{type}/{urn}?fields={keyNames}`>>
+* <<read3, Read - `POST /{type}/findByUrns?fields={keyNames}`>>
+* <<read4, Read - `POST /{type}/findByKeyValuePairs?page={pageNumber}&size={pageSize}`>>
+* <<delete1, Delete - `DELETE /{type}/{urn}`>>
+
+=== JSON Fields
+[width="100%",options="header",cols="1,^1,^1,^1,5"]
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String | generated, value and format depend on the persistence layer implementation | no default | will be generated if not provided
+| `urn` | String representation of a thing identifier in a common scheme, e.g. `urn:thing:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | no | will be generated if not provided
+| `type` | String | no | yes | the type of the Thing (e.g., "Employee", "Building")
+| `active` | boolean |  true | no |
+| `tenantUrn` | String |  generated | generated | The representation of the tenant identifier in a common scheme, e.g. `urn:tenant:uuid:<UUID>`. Only present in returned JSON Objects
+| [JSON metadata] | JSON | no default | false | Metadata for the thing, e.g. `name`, `description`, ..
```

Comment:
formatting here looks off because the text is not highlighted the same way.  What if we do
````
 ``[JSON metadata]`` 
`````